### DISCUSSION TITLE
Base Sales Calculation and Contribution Plot

### DISCUSSION THREAD

**Discussion by dangoML** [sibylhe/mmm_stan#54] [Q&A]
Hi! I first wanted to start off by saying thank you for sharing this! 

I was wondering how you go about creating the Media Contribution plot.
![image](https://user-images.githubusercontent.com/7134647/140113324-778abeeb-acce-4ef7-95c0-e3211d43230a.png)

I was able to produce it on my end by doing the following:
1) Weekly Contribution for each media = (Weekly ROAS of each media) x (Weekly Spend)
2) Base Sales = (Weekly Sales) - sum(Weekly Contribution for each media)
3) Plot all weekly values for each media and Base Sales
![image](https://user-images.githubusercontent.com/7134647/140151556-f4025ff7-8802-4497-ae2b-b45709605d74.png)
![image](https://user-images.githubusercontent.com/7134647/140155682-be6b50a0-351c-4845-a29e-9ec83df75173.png)

The problem is that this calculation of Base Sales yields a different value than the Base Sales derived from the control model. In fact, when conducting this calculation via ROAS some of the Base Sales values are negative!!

![image](https://user-images.githubusercontent.com/7134647/140153826-e0b35950-b917-4a1a-926a-8206d2db82b9.png)

I appreciate any insight you can provide.

**Comment by GrizzledLotus** [comment]
Check the file, "mmm_result_plot_template.xlsx", in the main branch. It seems to have tabs that match the csv output files from his code, [mcdf1, mc_pct, roas1y_df1]. Put your baseline sales contribution in the first column and your media spends in the subsequent columns. Be sure to adjust the 3 charts data ranges for your specific outcome.

For mc_pct, I'm unsure of which ROAS calculation to use, which comes from the roas1y_df1 data table. I used the overall ROAS avg value.

For the roas1y_df1, the chart uses two columns ROAS and mROAS, but the output from his code provides 4 columns (weekly ROAS mean and media, mROAS and overall ROAS).

**Reply by dangoML** [reply]
Hey! Thanks for getting back to me. I think maybe I didnt clearly write out my question.

What I am confused on are the actual values of Base Sales derived from the Control Model vs the MMM Model.
1) Why are they different?
2) The Control model also outputs MAPE, how do we interpret that value?
